### ISSUE TITLE
MNIST Stable HLO Conversion

### ISSUE THREAD

**Issue by AleksKnezevic** [tenstorrent/tt-mlir#428]
@mmanzoorTT , can you please generate an mnist (version without convolutions) stable hlo graph? Once done, @mrakitaTT can you please add the required ops to the converter.

**Comment by mrakitaTT** [comment]
Thank you Asif, looks good! I will continue adding the necessary ops to the converter.

Btw, I see that you've probably followed [this JAX tutorial](https://jax.readthedocs.io/en/latest/notebooks/Neural_Network_and_Data_Loading.html) where all ops are manually implemented. There is a google library for JAX called [Flax](https://github.com/google/flax) which already implements most of NN layers, so maybe it can save you some time when we scale up to add conv/maxpool layers and more. Just FYI.

**Comment by mrakitaTT** [comment]
@AleksKnezevic do we want to support only inference pass at first and then after that full training, or we need support for training already?

**Comment by mmanzoorTT** [comment]
Thanks @mrakitaTT, you are right. I am using Flax for my other experiments. I used JAX here to keep the code simple.

**Comment by staylorTT** [comment]
@mrakitaTT Is it safe to close this out now?

**Comment by mrakitaTT** [comment]
Yup, from tt-xla now both MNIST MLP and CNN tests pass end to end, closing.
